Content.Server/Animals/Systems/PettingSystem.cs

```diff
@@ -0,0 +1,74 @@
+using Content.Server.Popups;
+using Content.Shared.Audio;
+using Content.Shared.Interaction;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+using Content.Server.Animals.Components;
+using Content.Shared.MobState;
+using Robust.Shared.Random;
+
+namespace Content.Server.Animals.Systems;
+
+public class PettingSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PettableComponent, InteractHandEvent>(OnInteractHand);
+        SubscribeLocalEvent<PettableComponent, MobStateChangedEvent>(OnMobStateChanged);
+    }
+
+    private void OnMobStateChanged(EntityUid uid, PettableComponent component, MobStateChangedEvent args)
+    {
+        if (!args.Component.IsAlive()) // if not alive (dead, incapacitated, etc.)
+            component.PetSuccessChance = -1.0f; // set to the "invalid" value, which suppresses all popups and sound effects.
+
+        return;
+    }
```

Comment:
This never gets unset

Reply:
Good catch, will fix.

Reply:
Addressed and ported to InteractionPopupSystem